modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/Fragment.java

```diff
@@ -64,19 +73,21 @@ public class Fragment {
      * @param remotes Remote sources of the fragment.
      */
     public Fragment(long id, IgniteRel root, List<IgniteReceiver> remotes) {
-        this(id, root, remotes, null, null);
+        this(id, root, expectsCorrelatesFromOutside(root), remotes, null, null);
     }
 
     /**
      * Constructor.
      * TODO Documentation https://issues.apache.org/jira/browse/IGNITE-15859
      */
-    Fragment(long id, IgniteRel root, List<IgniteReceiver> remotes, @Nullable String rootSer, @Nullable FragmentMapping mapping) {
+    Fragment(long id, IgniteRel root, boolean expectsCorrelatesFromOutside, List<IgniteReceiver> remotes,
```

Comment:
I guess, we can pass 'expectsCorrelatesFromOutside' to constructor through FragmentProto in FragmenSplitter.go(Fragment).
see: stack.push(new FragmentProto(IdGenerator.nextId(), fragment.root()));

Reply:
I reworked this part because of a races (see comment to commit [74fe27e](https://github.com/apache/ignite-3/pull/1617/commits/74fe27ec36f980bc9fe3688deff68ca5a4efbfff) for details). Your suggestion was taken into account